--- /tmp/x86.rs	2024-04-25 23:47:26.697256628 +0000
+++ x86.rs	2024-04-25 23:47:33.253251743 +0000
@@ -11,8 +11,7 @@
 intrinsics! {
     #[naked]
     #[cfg(all(
-        windows,
-        target_env = "gnu",
+        any(all(windows, target_env = "gnu"), target_os = "uefi"),
         not(feature = "no-asm")
     ))]
     pub unsafe extern "C" fn ___chkstk_ms() {
@@ -41,8 +40,7 @@
     // FIXME: __alloca should be an alias to __chkstk
     #[naked]
     #[cfg(all(
-        windows,
-        target_env = "gnu",
+        any(all(windows, target_env = "gnu"), target_os = "uefi"),
         not(feature = "no-asm")
     ))]
     pub unsafe extern "C" fn __alloca() {
@@ -54,8 +52,7 @@

     #[naked]
     #[cfg(all(
-        windows,
-        target_env = "gnu",
+        any(all(windows, target_env = "gnu"), target_os = "uefi"),
         not(feature = "no-asm")
     ))]
     pub unsafe extern "C" fn ___chkstk() {
